lib/src/Base/Geom/Mesh.cxx

```diff
@@ -223,16 +223,16 @@ Bool Mesh::checkPointInSimplexWithCoordinates(const Point & point,
     const Scalar x02 = vertices_(simplices_(index, 2), 0) - x0;
     const Scalar y02 = vertices_(simplices_(index, 2), 1) - y0;
     const Scalar det = (x02 * y01 - y02 * x01);
-    const Scalar x = point[0] - x0;
-    const Scalar y = point[1] - y0;
     if (det == 0.0)
     {
       return false;
     }
+    const Scalar x = point[0] - x0;
+    const Scalar y = point[1] - y0;
     coordinates = Point(3);
     coordinates[1] = (x02 * y - y02 * x) / det;
     coordinates[2] = (x * y01 - y * x01) / det;
-    coordinates[0] = 1.0 - coordinates[1] - coordinates[2];
+    coordinates[0] = 0.5 + (0.5 - coordinates[1] - coordinates[2]);
```

Comment:
or maybe `(0.5 - coordinates[1]) + (0.5 - coordinates[2])`, but at the end I thin we will have to introduce an epsilon to check if the given point is inside of a slightly larger simplex